pkg/losetup/losetup_linux.go

```diff
@@ -0,0 +1,105 @@
+package losetup
+
+import (
+	"fmt"
+	"os"
+	"syscall"
+)
+
+const (
+	/*
+	 * IOCTL commands --- we will commandeer 0x4C ('L')
+	 */
+	LOOP_SET_CAPACITY = 0x4C07
+	LOOP_CHANGE_FD    = 0x4C06
+	LOOP_GET_STATUS64 = 0x4C05
+	LOOP_SET_STATUS64 = 0x4C04
+	LOOP_GET_STATUS   = 0x4C03
+	LOOP_SET_STATUS   = 0x4C02
+	LOOP_CLR_FD       = 0x4C01
+	LOOP_SET_FD       = 0x4C00
+	LO_NAME_SIZE      = 64
+	LO_KEY_SIZE       = 32
+	/* /dev/loop-control interface */
+	LOOP_CTL_ADD      = 0x4C80
+	LOOP_CTL_REMOVE   = 0x4C81
+	LOOP_CTL_GET_FREE = 0x4C82
+)
+
+// FindLoopDevice finds an unused loop device.
+func FindLoopDevice() (name string, err error) {
+	cfd, err := syscall.Open("/dev/loop-control", syscall.O_RDWR, 0)
+	if err != nil {
+		return "", err
+	}
+	defer syscall.Close(cfd)
+
+	if number, err := LoopCtlGetFree(uintptr(cfd)); err != nil {
+		return "", err
+	} else {
+		return fmt.Sprintf("/dev/loop%d", number), nil
+	}
+
+}
+
+// LoopClearFd clears the loop device associated with filedescriptor fd.
+func LoopClearFd(fd uintptr) error {
```

Comment:
Totally optional, but I'd just call this package loop, and remove the name Loop from all the functions. (you don't want them to stutter like `loop.LoopClearFd`, just `loop.ClearFd` would be nice)